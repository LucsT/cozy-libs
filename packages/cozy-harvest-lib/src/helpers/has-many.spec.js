const { updateHasManyItem } = require('./has-many')

describe('has-many helpers', () => {
  it('should work', () => {
    const doc = {
      _id: 'doc-1',
      label: 'Document 1',
      relationships: {
        author: { data: { _id: 'author-1', _type: 'io.cozy.authors' } },
        books: {
          data: [
            { _id: 'book-1', _type: 'io.cozy.books' },
            { _id: 'book-2', _type: 'io.cozy.books' },
            { _id: 'book-3', _type: 'io.cozy.books' }
          ]
        }
      }
    }

    const doc2 = updateHasManyItem(doc, 'books', 'book-2', rel => {
      return { ...rel, meta: { read: true } }
    })

    const doc3 = updateHasManyItem(doc2, 'books', 'book-3', rel => {
      return { ...rel, meta: { read: true } }
    })

    expect(doc2).toMatchSnapshot()
    expect(doc3).toMatchSnapshot()
  })
})
