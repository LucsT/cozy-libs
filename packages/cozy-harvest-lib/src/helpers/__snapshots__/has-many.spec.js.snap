// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has-many helpers should work 1`] = `
Object {
  "_id": "doc-1",
  "label": "Document 1",
  "relationships": Object {
    "author": Object {
      "data": Object {
        "_id": "author-1",
        "_type": "io.cozy.authors",
      },
    },
    "books": Object {
      "data": Array [
        Object {
          "_id": "book-1",
          "_type": "io.cozy.books",
        },
        Object {
          "_id": "book-2",
          "_type": "io.cozy.books",
          "metadata": Object {
            "read": true,
          },
        },
        Object {
          "_id": "book-3",
          "_type": "io.cozy.books",
        },
      ],
    },
  },
}
`;

exports[`has-many helpers should work 2`] = `
Object {
  "_id": "doc-1",
  "label": "Document 1",
  "relationships": Object {
    "author": Object {
      "data": Object {
        "_id": "author-1",
        "_type": "io.cozy.authors",
      },
    },
    "books": Object {
      "data": Array [
        Object {
          "_id": "book-1",
          "_type": "io.cozy.books",
        },
        Object {
          "_id": "book-2",
          "_type": "io.cozy.books",
          "metadata": Object {
            "read": true,
          },
        },
        Object {
          "_id": "book-3",
          "_type": "io.cozy.books",
          "metadata": Object {
            "read": true,
          },
        },
      ],
    },
  },
}
`;
